Use yarn workspaces

FredyC opened the pull request:
Closes #140 

No serious changes, except I've moved the `@types/react` to a root package.json. This package is very fragile in regards to duplicate installations. Had so many issues with it in the past and it's not worth it to keep it fragmented over packages.

It's still possible to use `yarn bootstrap` so I've kept information about in contributing file, but running `yarn` only works the same way now.

I haven't provided the updated root yarn.lock. It's kinda recommended for maintainers to always generate that because of security concerns. In case you trust me enough, I can add it as well, there is a lot of changes in it.

Btw, any reason why you have decided to keep exact versions of packages in package.json files? I mean that's the job for the lock file to keep the exact version around. To be honest, I have been doing it the same way in the past, but then I've discovered the magical `yarn upgradeInteractive` command which makes things so much easier. But it does work with semver ranges only, otherwise, it won't see any updates available.

FredyC commented on packages/react-google-maps-api-storybook/package.json:
```diff
@@ -20,7 +20,7 @@
   },
   "dependencies": {
-    "@react-google-maps/api": "1.2.3-alpha.12",
+    "@react-google-maps/api": "1.2.3",
```
alpha.12 was never published and it fails on installation

FredyC commented:
Hm, that's strange why is build failing ...

> /usr/local/bin/build: line 34: yarn: command not found

JustFly1984 commented:
Have you tried to build docs locally? I see TS errors and missing `@react-google-maps/marker-clusterer

I have been updating @react-google-maps/api@1.2.3 in own app and it works.

JustFly1984 commented:
this is a log from netlify:
```
12:04:28 PM: warning workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/gatsby-example > @babel/polyfill@7.4.4: 🚨 As of Babel 7.4.0, this
12:04:28 PM: package has been deprecated in favor of directly
12:04:28 PM: including core-js/stable (to polyfill ECMAScript
12:04:28 PM: features) and regenerator-runtime/runtime
12:04:28 PM: (needed to use transpiled generator functions):
12:04:28 PM:   > import "core-js/stable";
12:04:28 PM:   > import "regenerator-runtime/runtime";
12:04:28 PM: warning workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/gatsby-example > gatsby > @babel/polyfill@7.4.4: 🚨 As of Babel 7.4.0, this
12:04:28 PM: package has been deprecated in favor of directly
12:04:28 PM: including core-js/stable (to polyfill ECMAScript
12:04:28 PM: features) and regenerator-runtime/runtime
12:04:28 PM: (needed to use transpiled generator functions):
12:04:28 PM:   > import "core-js/stable";
12:04:28 PM:   > import "regenerator-runtime/runtime";
12:04:29 PM: warning workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-testing-library@7.0.1: 🚨  react-testing-library has moved to @testing-library/react. Please uninstall react-testing-library and install @testing-library/react instead, or use an older version of react-testing-library. Learn more about this change here: https://github.com/testing-library/dom-testing-library/issues/260 Thanks! :)
12:04:30 PM: warning workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/gatsby-example > gatsby > joi@14.3.1: This module has moved and is now available at @hapi/joi. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
12:04:31 PM: warning workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-testing-library > dom-testing-library@4.1.1: 🚨  dom-testing-library has moved to @testing-library/dom. Please uninstall dom-testing-library and install @testing-library/dom instead, or use an older version of dom-testing-library. Learn more about this change here: https://github.com/testing-library/dom-testing-library/issues/260 Thanks! :)
12:04:31 PM: warning workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/gatsby-example > gatsby-plugin-favicon > html-react-parser > react-dom-core > react > fbjs > core-js@1.2.7: core-js@<2.6.8 is no longer maintained. Please, upgrade to core-js@3 or at least to actual version of core-js@2.
12:04:32 PM: warning workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/gatsby-example > gatsby > joi > hoek@6.1.3: This module has moved and is now available at @hapi/hoek. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
12:04:32 PM: warning workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/gatsby-example > gatsby > joi > topo@3.0.3: This module has moved and is now available at @hapi/topo. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
12:04:34 PM: warning workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/storybook > eslint > file-entry-cache > flat-cache > circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.
12:04:36 PM: [2/4] Fetching packages...
12:05:15 PM: info fsevents@1.2.7: The platform "linux" is incompatible with this module.
12:05:15 PM: info "fsevents@1.2.7" is an optional dependency and failed compatibility check. Excluding it from installation.
12:05:15 PM: [3/4] Linking dependencies...
12:05:15 PM: warning " > ts-jest@24.0.2" has unmet peer dependency "jest@>=24 <25".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api@1.2.3" has unmet peer dependency "react@^16.6.3".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api@1.2.3" has unmet peer dependency "react-dom@^16.6.3".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/gatsby-example > stylelint-css-modules@0.8.3" has incorrect peer dependency "stylelint@^9.0.0".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/marker-clusterer > eslint-import-resolver-typescript@1.1.1" has unmet peer dependency "eslint@*".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/marker-clusterer > eslint-import-resolver-typescript@1.1.1" has unmet peer dependency "eslint-plugin-import@*".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/marker-clusterer > awesome-typescript-loader@5.2.1" has unmet peer dependency "typescript@^2.7 || ^3".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-testing-library@7.0.1" has unmet peer dependency "react@*".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-testing-library@7.0.1" has unmet peer dependency "react-dom@*".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/marker-clusterer > @typescript-eslint/parser@1.9.0" has unmet peer dependency "eslint@^5.0.0".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist@9.1.2" has unmet peer dependency "react@>=15".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist@9.1.2" has unmet peer dependency "react-dom@>=15".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist > markdown-to-jsx@6.10.2" has unmet peer dependency "react@>= 0.14.0".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist > react-group@3.0.0" has unmet peer dependency "react@>=16.0.0".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist > react-icons@2.2.7" has unmet peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist > react-icons@2.2.7" has unmet peer dependency "react-dom@^0.14.0 || ^15.0.0 || ^16.0.0".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist > react-simple-code-editor@0.9.10" has unmet peer dependency "react@^16.0.0".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist > react-simple-code-editor@0.9.10" has unmet peer dependency "react-dom@^16.0.0".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist > react-icons > react-icon-base@2.1.0" has unmet peer dependency "react@*".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist > react-icons > react-icon-base@2.1.0" has unmet peer dependency "react-dom@*".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/api > react-styleguidist > react-icons > react-icon-base@2.1.0" has unmet peer dependency "prop-types@*".
12:05:15 PM: warning "workspace-aggregator-04792980-e448-4f6c-9cf1-fac0e988d9fc > @react-google-maps/storybook > @storybook/react > @storybook/core > @storybook/ui > react-treebeard@3.2.3" has incorrect peer dependency "@emotion/styled@^10.0.10".
12:05:49 PM: [4/4] Building fresh packages...
12:06:01 PM: success Saved lockfile.
12:06:01 PM: Done in 94.30s.
12:06:01 PM: NPM modules installed using Yarn
12:06:02 PM: Started restoring cached go cache
12:06:04 PM: Finished restoring cached go cache
12:06:04 PM: Installing Go version 1.10
12:06:04 PM: unset GOOS;
12:06:04 PM: unset GOARCH;
12:06:04 PM: export GOROOT='/opt/buildhome/.gimme_cache/versions/go1.10.linux.amd64';
12:06:04 PM: export PATH="/opt/buildhome/.gimme_cache/versions/go1.10.linux.amd64/bin:${PATH}";
12:06:04 PM: go version >&2;
12:06:04 PM: export GIMME_ENV='/opt/buildhome/.gimme_cache/env/go1.10.linux.amd64.env';
12:06:04 PM: go version go1.10 linux/amd64
12:06:04 PM: Installing missing commands
12:06:04 PM: Verify run directory
12:06:04 PM: Executing user command: yarn build:docs
12:06:04 PM: yarn run v1.3.2
12:06:04 PM: $ cd packages/react-google-maps-api && yarn && yarn docs:build && cd ../..
12:06:05 PM: [1/4] Resolving packages...
12:06:08 PM: success Already up-to-date.
12:06:09 PM: $ npx styleguidist build --config ./styleguide.config.js --verbose
12:06:10 PM: Styleguidist config:
12:06:10 PM: {
12:06:10 PM:   assetsDir: undefined,
12:06:10 PM:   compilerConfig: {
12:06:10 PM:     objectAssign: 'Object.assign',
12:06:10 PM:     target: {
12:06:10 PM:       ie: 11
12:06:10 PM:     },
12:06:10 PM:     transforms: {
12:06:10 PM:       modules: false,
12:06:10 PM:       dangerousTaggedTemplateString: true
12:06:10 PM:     }
12:06:10 PM:   },
12:06:10 PM:   components: undefined,
12:06:10 PM:   configDir: '/opt/build/repo/packages/react-google-maps-api',
12:06:10 PM:   context: {},
12:06:10 PM:   contextDependencies: undefined,
12:06:10 PM:   configureServer: undefined,
12:06:10 PM:   dangerouslyUpdateWebpackConfig: undefined,
12:06:10 PM:   defaultExample: false,
12:06:10 PM:   exampleMode: 'collapse',
12:06:10 PM:   getComponentPathLine: Function getComponentPathLine,
12:06:10 PM:   getExampleFilename: Function default,
12:06:10 PM:   handlers: componentPath => reactDocgen.defaultHandlers.concat(createDisplayNameHandler(componentPath)),
12:06:10 PM:   ignore: [
12:06:10 PM:     '**/*.js',
12:06:10 PM:     '**/*.ts'
12:06:10 PM:   ],
12:06:10 PM:   editorConfig: undefined,
12:06:10 PM:   logger: undefined,
12:06:10 PM:   moduleAliases: undefined,
12:06:10 PM:   mountPointId: 'rsg-root',
12:06:10 PM:   pagePerSection: false,
12:06:10 PM:   previewDelay: 500,
12:06:10 PM:   printBuildInstructions: undefined,
12:06:10 PM:   printServerInstructions: undefined,
12:06:10 PM:   propsParser: Function parse,
12:06:10 PM:   require: [],
12:06:10 PM:   resolver: Function default,
12:06:10 PM:   ribbon: undefined,
12:06:10 PM:   sections: [
12:06:10 PM:     {
12:06:10 PM:       name: 'Introduction',
12:06:10 PM:       content: 'src/docs/introduction.md'
12:06:10 PM:     },
12:06:10 PM:     {
12:06:10 PM:       name: 'Getting Started',
12:06:10 PM:       content: 'src/docs/getting-started.md'
12:06:10 PM:     },
12:06:10 PM:     {
12:06:10 PM:       name: 'Components',
12:06:10 PM:       components: [
12:06:10 PM:         'src/LoadScript.tsx',
12:06:10 PM:         'src/GoogleMap.tsx',
12:06:10 PM:         'src/components/**/*.tsx'
12:06:10 PM:       ]
12:06:10 PM:     }
12:06:10 PM:   ],
12:06:10 PM:   serverHost: '0.0.0.0',
12:06:10 PM:   serverPort: 6060,
12:06:10 PM:   showCode: false,
12:06:10 PM:   showUsage: false,
12:06:10 PM:   showSidebar: true,
12:06:10 PM:   skipComponentsWithoutExample: false,
12:06:10 PM:   sortProps: undefined,
12:06:10 PM:   styleguideComponents: undefined,
12:06:10 PM:   styleguideDir: '/opt/build/repo/packages/react-google-maps-api/styleguide',
12:06:10 PM:   styles: {},
12:06:10 PM:   template: {},
12:06:10 PM:   theme: {},
12:06:10 PM:   title: 'React Google Maps Api Style Guide',
12:06:10 PM:   updateDocs: undefined,
12:06:10 PM:   updateExample: Function default,
12:06:10 PM:   updateWebpackConfig: undefined,
12:06:10 PM:   usageMode: 'expand',
12:06:10 PM:   verbose: true,
12:06:10 PM:   version: undefined,
12:06:10 PM:   webpackConfig: {
12:06:10 PM:     module: {
12:06:10 PM:       rules: [
12:06:10 PM:         {
12:06:10 PM:           test: /\.ts|\.tsx$/,
12:06:10 PM:           loader: 'awesome-typescript-loader'
12:06:10 PM:         }
12:06:10 PM:       ]
12:06:10 PM:     },
12:06:10 PM:     resolve: {
12:06:10 PM:       extensions: [
12:06:10 PM:         '.ts',
12:06:10 PM:         '.tsx',
12:06:10 PM:         '.js'
12:06:10 PM:       ]
12:06:10 PM:     }
12:06:10 PM:   }
12:06:10 PM: }
12:06:10 PM: Building style guide...
12:06:11 PM: clean-webpack-plugin: /opt/build/repo/packages/react-google-maps-api/styleguide/build has been removed.
12:06:11 PM: Webpack config:
12:06:11 PM: {
12:06:11 PM:   entry: [
12:06:11 PM:     '/opt/build/repo/node_modules/react-styleguidist/lib/client/index'
12:06:11 PM:   ],
12:06:11 PM:   mode: 'production',
12:06:11 PM:   output: {
12:06:11 PM:     path: '/opt/build/repo/packages/react-google-maps-api/styleguide',
12:06:11 PM:     filename: 'build/bundle.[chunkhash:8].js',
12:06:11 PM:     chunkFilename: 'build/[name].[chunkhash:8].js',
12:06:11 PM:     publicPath: '',
12:06:11 PM:     webassemblyModuleFilename: '[modulehash].module.wasm',
12:06:11 PM:     library: '',
12:06:11 PM:     hotUpdateFunction: 'webpackHotUpdate',
12:06:11 PM:     jsonpFunction: 'webpackJsonp',
12:06:11 PM:     chunkCallbackName: 'webpackChunk',
12:06:11 PM:     globalObject: 'window',
12:06:11 PM:     devtoolNamespace: '',
12:06:11 PM:     libraryTarget: 'var',
12:06:11 PM:     pathinfo: false,
12:06:11 PM:     sourceMapFilename: '[file].map[query]',
12:06:11 PM:     hotUpdateChunkFilename: '[id].[hash].hot-update.js',
12:06:11 PM:     hotUpdateMainFilename: '[hash].hot-update.json',
12:06:11 PM:     crossOriginLoading: false,
12:06:11 PM:     jsonpScriptType: false,
12:06:11 PM:     chunkLoadTimeout: 120000,
12:06:11 PM:     hashFunction: 'md4',
12:06:11 PM:     hashDigest: 'hex',
12:06:11 PM:     hashDigestLength: 20,
12:06:11 PM:     devtoolLineToLine: false,
12:06:11 PM:     strictModuleExceptionHandling: false
12:06:11 PM:   },
12:06:11 PM:   resolve: {
12:06:11 PM:     extensions: [
12:06:11 PM:       '.js',
12:06:11 PM:       '.jsx',
12:06:11 PM:       '.json',
12:06:11 PM:       '.ts',
12:06:11 PM:       '.tsx',
12:06:11 PM:       '.js'
12:06:11 PM:     ],
12:06:11 PM:     alias: {
12:06:11 PM:       'rsg-components': '/opt/build/repo/node_modules/react-styleguidist/lib/client/rsg-components'
12:06:11 PM:     },
12:06:11 PM:     unsafeCache: true,
12:06:11 PM:     modules: [
12:06:11 PM:       'node_modules'
12:06:11 PM:     ],
12:06:11 PM:     mainFiles: [
12:06:11 PM:       'index'
12:06:11 PM:     ],
12:06:11 PM:     aliasFields: [
12:06:11 PM:       'browser'
12:06:11 PM:     ],
12:06:11 PM:     mainFields: [
12:06:11 PM:       'browser',
12:06:11 PM:       'module',
12:06:11 PM:       'main'
12:06:11 PM:     ],
12:06:11 PM:     cacheWithContext: false
12:06:11 PM:   },
12:06:11 PM:   plugins: [
12:06:11 PM:     StyleguidistOptionsPlugin {
12:06:11 PM:       options: Object {47},
12:06:11 PM:       plugin: function () { [native code] }
12:06:11 PM:     },
12:06:11 PM:     MiniHtmlWebpackPlugin {
12:06:11 PM:       options: {
12:06:11 PM:         context: {
12:06:11 PM:           title: 'React Google Maps Api Style Guide',
12:06:11 PM:           container: 'rsg-root',
12:06:11 PM:           trimWhitespace: true
12:06:11 PM:         },
12:06:11 PM:         template: Function template
12:06:11 PM:       },
12:06:11 PM:       plugin: function () { [native code] }
12:06:11 PM:     },
12:06:11 PM:     DefinePlugin {
12:06:11 PM:       definitions: {
12:06:11 PM:         'process.env.NODE_ENV': '"production"',
12:06:11 PM:         'process.env.STYLEGUIDIST_ENV': '"production"'
12:06:11 PM:       }
12:06:11 PM:     },
12:06:11 PM:     CleanWebpackPlugin {
12:06:11 PM:       paths: [
12:06:11 PM:         'build'
12:06:11 PM:       ],
12:06:11 PM:       options: {
12:06:11 PM:         root: '/opt/build/repo/packages/react-google-maps-api/styleguide',
12:06:11 PM:         verbose: true,
12:06:11 PM:         allowExternal: false,
12:06:11 PM:         dry: false
12:06:11 PM:       }
12:06:11 PM:     },
12:06:11 PM:     {
12:06:11 PM:       apply: Function apply
12:06:11 PM:     }
12:06:11 PM:   ],
12:06:11 PM:   performance: {
12:06:11 PM:     hints: false,
12:06:11 PM:     maxAssetSize: 250000,
12:06:11 PM:     maxEntrypointSize: 250000
12:06:11 PM:   },
12:06:11 PM:   optimization: {
12:06:11 PM:     minimizer: [
12:06:11 PM:       TerserPlugin {
12:06:11 PM:         options: {
12:06:11 PM:           test: /\.m?js(\?.*)?$/i,
12:06:11 PM:           chunkFilter: () => true,
12:06:11 PM:           warningsFilter: () => true,
12:06:11 PM:           extractComments: false,
12:06:11 PM:           sourceMap: false,
12:06:11 PM:           cache: true,
12:06:11 PM:           cacheKeys: defaultCacheKeys => defaultCacheKeys,
12:06:11 PM:           parallel: true,
12:06:11 PM:           include: undefined,
12:06:11 PM:           exclude: undefined,
12:06:11 PM:           minify: undefined,
12:06:11 PM:           terserOptions: {
12:06:11 PM:             output: {
12:06:11 PM:               comments: /^\**!|@preserve|@license|@cc_on/i
12:06:11 PM:             },
12:06:11 PM:             ie8: false,
12:06:11 PM:             ecma: 5,
12:06:11 PM:             compress: {
12:06:11 PM:               keep_fnames: true,
12:06:11 PM:               warnings: false,
12:06:11 PM:               reduce_funcs: false
12:06:11 PM:             },
12:06:11 PM:             mangle: {
12:06:11 PM:               keep_fnames: true
12:06:11 PM:             }
12:06:11 PM:           }
12:06:11 PM:         }
12:06:11 PM:       }
12:06:11 PM:     ],
12:06:11 PM:     removeAvailableModules: true,
12:06:11 PM:     removeEmptyChunks: true,
12:06:11 PM:     mergeDuplicateChunks: true,
12:06:11 PM:     flagIncludedChunks: true,
12:06:11 PM:     occurrenceOrder: true,
12:06:11 PM:     sideEffects: true,
12:06:11 PM:     providedExports: true,
12:06:11 PM:     usedExports: true,
12:06:11 PM:     concatenateModules: true,
12:06:11 PM:     splitChunks: {
12:06:11 PM:       hidePathInfo: true,
12:06:11 PM:       chunks: 'async',
12:06:11 PM:       minSize: 30000,
12:06:11 PM:       minChunks: 1,
12:06:11 PM:       maxAsyncRequests: 5,
12:06:11 PM:       automaticNameDelimiter: '~',
12:06:11 PM:       maxInitialRequests: 3,
12:06:11 PM:       name: true,
12:06:11 PM:       cacheGroups: {
12:06:11 PM:         default: {
12:06:11 PM:           automaticNamePrefix: '',
12:06:11 PM:           reuseExistingChunk: true,
12:06:11 PM:           minChunks: 2,
12:06:11 PM:           priority: -20
12:06:11 PM:         },
12:06:11 PM:         vendors: {
12:06:11 PM:           automaticNamePrefix: 'vendors',
12:06:11 PM:           test: /[\\\/]node_modules[\\\/]/,
12:06:11 PM:           priority: -10
12:06:11 PM:         }
12:06:11 PM:       }
12:06:11 PM:     },
12:06:11 PM:     runtimeChunk: undefined,
12:06:11 PM:     noEmitOnErrors: true,
12:06:11 PM:     checkWasmTypes: true,
12:06:11 PM:     mangleWasmImports: false,
12:06:11 PM:     namedModules: false,
12:06:11 PM:     hashedModuleIds: false,
12:06:11 PM:     namedChunks: false,
12:06:11 PM:     portableRecords: false,
12:06:11 PM:     minimize: true,
12:06:11 PM:     nodeEnv: 'production'
12:06:11 PM:   },
12:06:11 PM:   module: {
12:06:11 PM:     rules: [
12:06:11 PM:       {
12:06:11 PM:         test: /\.ts|\.tsx$/,
12:06:11 PM:         loader: 'awesome-typescript-loader'
12:06:11 PM:       }
12:06:11 PM:     ],
12:06:11 PM:     unknownContextRequest: '.',
12:06:11 PM:     unknownContextRegExp: false,
12:06:11 PM:     unknownContextRecursive: true,
12:06:11 PM:     unknownContextCritical: true,
12:06:11 PM:     exprContextRequest: '.',
12:06:11 PM:     exprContextRegExp: false,
12:06:11 PM:     exprContextRecursive: true,
12:06:11 PM:     exprContextCritical: true,
12:06:11 PM:     wrappedContextRegExp: /.*/,
12:06:11 PM:     wrappedContextRecursive: true,
12:06:11 PM:     wrappedContextCritical: false,
12:06:11 PM:     strictExportPresence: false,
12:06:11 PM:     strictThisContextOnImports: false,
12:06:11 PM:     unsafeCache: false,
12:06:11 PM:     defaultRules: [
12:06:11 PM:       {
12:06:11 PM:         type: 'javascript/auto',
12:06:11 PM:         resolve: {}
12:06:11 PM:       },
12:06:11 PM:       {
12:06:11 PM:         test: /\.mjs$/i,
12:06:11 PM:         type: 'javascript/esm',
12:06:11 PM:         resolve: {
12:06:11 PM:           mainFields: [
12:06:11 PM:             'browser',
12:06:11 PM:             'main'
12:06:11 PM:           ]
12:06:11 PM:         }
12:06:11 PM:       },
12:06:11 PM:       {
12:06:11 PM:         test: /\.json$/i,
12:06:11 PM:         type: 'json'
12:06:11 PM:       },
12:06:11 PM:       {
12:06:11 PM:         test: /\.wasm$/i,
12:06:11 PM:         type: 'webassembly/experimental'
12:06:11 PM:       }
12:06:11 PM:     ]
12:06:11 PM:   },
12:06:11 PM:   devtool: false,
12:06:11 PM:   cache: false,
12:06:11 PM:   context: '/opt/build/repo/packages/react-google-maps-api',
12:06:11 PM:   target: 'web',
12:06:11 PM:   node: {
12:06:11 PM:     console: false,
12:06:11 PM:     process: true,
12:06:11 PM:     global: true,
12:06:11 PM:     Buffer: true,
12:06:11 PM:     setImmediate: true,
12:06:11 PM:     __filename: 'mock',
12:06:11 PM:     __dirname: 'mock'
12:06:11 PM:   },
12:06:11 PM:   resolveLoader: {
12:06:11 PM:     unsafeCache: true,
12:06:11 PM:     mainFields: [
12:06:11 PM:       'loader',
12:06:11 PM:       'main'
12:06:11 PM:     ],
12:06:11 PM:     extensions: [
12:06:11 PM:       '.js',
12:06:11 PM:       '.json'
12:06:11 PM:     ],
12:06:11 PM:     mainFiles: [
12:06:11 PM:       'index'
12:06:11 PM:     ],
12:06:11 PM:     cacheWithContext: false
12:06:11 PM:   }
12:06:11 PM: }
12:06:11 PM: Loading components:
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/LoadScript.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/GoogleMap.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/addons/MarkerClusterer.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/directions/DirectionsRenderer.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/directions/DirectionsService.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/dom/OverlayView.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/drawing/Circle.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/drawing/Data.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/drawing/DrawingManager.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/drawing/InfoWindow.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/drawing/Marker.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/drawing/Polygon.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/drawing/Polyline.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/drawing/Rectangle.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/heatmap/HeatmapLayer.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/kml/KmlLayer.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/maps/BicyclingLayer.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/maps/TrafficLayer.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/overlays/GroundOverlay.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/places/Autocomplete.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/places/StandaloneSearchBox.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/streetview/StreetViewPanorama.tsx
12:06:11 PM: /opt/build/repo/packages/react-google-maps-api/src/components/streetview/StreetViewService.tsx
12:06:30 PM: ℹ ｢atl｣: Using typescript@3.4.5 from typescript
12:06:30 PM: ℹ ｢atl｣: Using tsconfig.json from /opt/build/repo/packages/react-google-maps-api/tsconfig.json
12:07:10 PM: ℹ ｢atl｣: Checking started in a separate process...
12:07:11 PM: ✖ ｢atl｣: Checking finished with 29 errors
12:07:11 PM:  FAIL  Failed to compile
12:07:11 PM: Module not found: Error: Can't resolve '@react-google-maps/marker-clusterer' in '/opt/build/repo/packages/react-google-maps-api/src/components/addons'
12:07:11 PM: [at-loader] ./src/components/drawing/Marker.tsx:10:27
12:07:11 PM:     TS2307: Cannot find module '@react-google-maps/marker-clusterer'.
12:07:11 PM: [at-loader] ./src/components/addons/MarkerClusterer.tsx:17:8
12:07:11 PM:     TS2307: Cannot find module '@react-google-maps/marker-clusterer'.
12:07:11 PM: [at-loader] ./src/__tests__/components/circle.test.tsx:22:1
12:07:11 PM:     TS2304: Cannot find name 'afterEach'.
12:07:11 PM: [at-loader] ./src/__tests__/components/circle.test.tsx:27:1
12:07:11 PM:     TS2593: Cannot find name 'describe'. Do you need to install type definitions for a test runner? Try `npm i @types/jest` or `npm i @types/mocha` and then add `jest` or `mocha` to the types field in your tsconfig.
12:07:11 PM: [at-loader] ./src/__tests__/components/circle.test.tsx:28:3
12:07:11 PM:     TS2593: Cannot find name 'it'. Do you need to install type definitions for a test runner? Try `npm i @types/jest` or `npm i @types/mocha` and then add `jest` or `mocha` to the types field in your tsconfig.
12:07:11 PM: [at-loader] ./src/__tests__/components/circle.test.tsx:29:24
12:07:11 PM:     TS2304: Cannot find name 'jest'.
12:07:11 PM: [at-loader] ./src/__tests__/components/circle.test.tsx:36:5
12:07:11 PM:     TS2304: Cannot find name 'expect'.
12:07:11 PM: [at-loader] ./src/__tests__/components/circle.test.tsx:39:3
12:07:11 PM:     TS2593: Cannot find name 'it'. Do you need to install type definitions for a test runner? Try `npm i @types/jest` or `npm i @types/mocha` and then add `jest` or `mocha` to the types field in your tsconfig.
12:07:11 PM: [at-loader] ./src/__tests__/components/circle.test.tsx:46:5
12:07:11 PM:     TS2304: Cannot find name 'expect'.
12:07:11 PM: [at-loader] ./src/__tests__/components/circle.test.tsx:49:3
12:07:11 PM:     TS2593: Cannot find name 'it'. Do you need to install type definitions for a test runner? Try `npm i @types/jest` or `npm i @types/mocha` and then add `jest` or `mocha` to the types field in your tsconfig.
12:07:11 PM: [at-loader] ./src/__tests__/components/circle.test.tsx:63:5
12:07:11 PM:     TS2304: Cannot find name 'expect'.
12:07:11 PM: [at-loader] ./src/__tests__/utils/helper.test.ts:4:11
12:07:11 PM:     TS2304: Cannot find name 'jest'.
12:07:11 PM: [at-loader] ./src/__tests__/utils/helper.test.ts:5:11
12:07:11 PM:     TS2304: Cannot find name 'jest'.
12:07:11 PM: [at-loader] ./src/__tests__/utils/helper.test.ts:8:1
12:07:11 PM:     TS2593: Cannot find name 'describe'. Do you need to install type definitions for a test runner? Try `npm i @types/jest` or `npm i @types/mocha` and then add `jest` or `mocha` to the types field in your tsconfig.
12:07:11 PM: [at-loader] ./src/__tests__/utils/helper.test.ts:9:3
12:07:11 PM:     TS2593: Cannot find name 'describe'. Do you need to install type definitions for a test runner? Try `npm i @types/jest` or `npm i @types/mocha` and then add `jest` or `mocha` to the types field in your tsconfig.
12:07:11 PM: [at-loader] ./src/__tests__/utils/helper.test.ts:10:5
12:07:11 PM:     TS2593: Cannot find name 'it'. Do you need to install type definitions for a test runner? Try `npm i @types/jest` or `npm i @types/mocha` and then add `jest` or `mocha` to the types field in your tsconfig.
12:07:11 PM: [at-loader] ./src/__tests__/utils/helper.test.ts:13:7
12:07:11 PM:     TS2304: Cannot find name 'expect'.
12:07:11 PM: [at-loader] ./src/__tests__/utils/helper.test.ts:14:7
12:07:11 PM:     TS2304: Cannot find name 'expect'.
12:07:11 PM: [at-loader] ../../node_modules/@jest/environment/build/index.d.ts:11:8
12:07:11 PM:     TS1192: Module '"/opt/build/repo/node_modules/jest-mock/build/index"' has no default export.
12:07:11 PM: [at-loader] ../../node_modules/jest-haste-map/build/types.d.ts:8:8
12:07:11 PM:     TS1192: Module '"fs"' has no default export.
12:07:11 PM: [at-loader] ../../node_modules/jest-haste-map/build/index.d.ts:8:8
12:07:11 PM:     TS1192: Module '"events"' has no default export.
12:07:11 PM: [at-loader] ../../node_modules/jest-runtime/build/types.d.ts:8:8
12:07:11 PM:     TS1192: Module '"/opt/build/repo/node_modules/jest-resolve/build/index"' has no default export.
12:07:11 PM: [at-loader] ../../node_modules/jest-runtime/build/index.d.ts:11:8
12:07:11 PM:     TS1192: Module '"/opt/build/repo/node_modules/jest-haste-map/build/index"' has no default export.
12:07:11 PM: [at-loader] ../../node_modules/jest-runtime/build/index.d.ts:12:8
12:07:11 PM:     TS1192: Module '"/opt/build/repo/node_modules/jest-resolve/build/index"' has no default export.
12:07:11 PM: [at-loader] ../../node_modules/jest-runner/build/types.d.ts:13:8
12:07:11 PM:     TS1192: Module '"/opt/build/repo/node_modules/jest-resolve/build/index"' has no default export.
12:07:11 PM: [at-loader] ../../node_modules/jest-runner/build/types.d.ts:14:8
12:07:11 PM:     TS1192: Module '"/opt/build/repo/node_modules/jest-runtime/build/index"' has no default export.
12:07:11 PM: [at-loader] ../../node_modules/@jest/reporters/build/types.d.ts:11:8
12:07:11 PM:     TS1192: Module '"/opt/build/repo/node_modules/jest-resolve/build/index"' has no default export.
12:07:11 PM: [at-loader] ../../node_modules/@jest/reporters/build/types.d.ts:12:8
12:07:11 PM:     TS1192: Module '"/opt/build/repo/node_modules/jest-runtime/build/index"' has no default export.
12:07:11 PM: [at-loader] ../../node_modules/jest-cli/build/cli/index.d.ts:10:71
12:07:11 PM:     TS2503: Cannot find namespace 'yargs'.
12:07:11 PM: error Command failed with exit code 1.
12:07:11 PM: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
12:07:11 PM: error Command failed with exit code 1.
12:07:11 PM: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
12:07:11 PM: Skipping functions preparation step: no functions directory set
12:07:11 PM: Caching artifacts
12:07:11 PM: Started saving node modules
12:07:11 PM: Finished saving node modules
12:07:11 PM: Started saving yarn cache
12:07:11 PM: Finished saving yarn cache
12:07:11 PM: Started saving pip cache
12:07:13 PM: Finished saving pip cache
12:07:13 PM: Started saving emacs cask dependencies
12:07:13 PM: Finished saving emacs cask dependencies
12:07:13 PM: Started saving maven dependencies
12:07:13 PM: Finished saving maven dependencies
12:07:13 PM: Started saving boot dependencies
12:07:13 PM: Finished saving boot dependencies
12:07:13 PM: Started saving go dependencies
12:07:14 PM: Finished saving go dependencies
12:07:14 PM: Error running command: Build script returned non-zero exit code: 1
12:07:14 PM: Failing build: Failed to build site
12:07:14 PM: failed during stage 'building site': Build script returned non-zero exit code: 1
12:07:15 PM: Finished processing build request in 3m7.829970405s
12:07:15 PM: Shutting down logging, 0 messages pending
```

JustFly1984 commented:
can you update typescript to 3.5.1? I think I have updated it before, but netlify uses 3.4.5 for some reason

JustFly1984 commented:
ℹ ｢atl｣: Using typescript@3.4.5 from typescript
 ℹ ｢atl｣: Using tsconfig.json from /opt/build/repo/packages/react-google-maps-api/tsconfig.json

I think the issues related to this lines

FredyC commented:
Yeah it works perfectly locally, but I am not Windows, so there might something I am missing. Perhaps it's related to not including `yarn.lock`? I will first try upgrading TS and then include lock if it doesn't help.

JustFly1984 commented:
you can commit your lock file safely, I will rebuild it anyway before publishing

FredyC commented:
Hm, apparently that either of that was a problem, but I don't see those previous errors in logs, there is only that cryptic one ... 

> 1:09:20 PM: Executing user command: yarn && yarn build
> 1:09:20 PM: /usr/local/bin/build: line 34: yarn: command not found

There is also this strange line like it would be installing deps with NPM...

> 1:09:17 PM: npm notice created a lockfile as package-lock.json. You should commit this file.

JustFly1984 commented:
We are also not using Windows, and netlify doesn't use Windows. why Windows?

It does strangely installs packages with NPM instead of yarn

>>>Starting today, if you deploy a site that includes a yarn.lock file in the base of the repository, we’ll use Yarn to install dependencies. 

https://www.netlify.com/blog/2016/11/01/yarn-support-on-netlify/

JustFly1984 commented:
Oh, I get it, I think you did not include yarn.lock for root

FredyC commented:
Well, `yarn.lock` was there before for sure, although outdated. From those logs, it seems that gatsby package is trying to install dependencies with NPM. I am not familiar with gatsby at all. Is it because I've deleted `yarn.lock` from that package subfolder? Is there some routine for gatsby that's installing deps separately?

Either way, I've rebased it to master and resolved conflicts. And yet it's still failing :/

FredyC commented:
Hm, I am out of ideas now. It's rebased on the latest master which is apparently building just fine, but only a couple of changes for this PR breaks it.

FredyC commented:
Oh, I have rebased PR #143 and it's actually failing the same way. How is that possible? There are only code changes 😵

JustFly1984 commented:
Installing NPM modules using NPM version 6.5.0

this means something wrong with yarn.lock files in the root of the repo.
also there is no need in lerna, so you can safely remove it.

FredyC commented:
> also there is no need in lerna, so you can safely remove it.

```
lerna notice cli v3.14.1
lerna ERR! EWORKSPACES Yarn workspaces are configured in package.json, but not enabled in lerna.json!
lerna ERR! EWORKSPACES Please choose one: useWorkspaces = true in lerna.json, or remove package.json workspaces config
```

It is needed ;)

Tried the revert, but it did not even updated `yarn.lock` and build is still failing, so that's not it. Can Netlify have an actual issue with Yarn workspaces? That would be rather silly.

FredyC commented:
Btw, where are you seeing it's installing with NPM?

> 5:09:26 PM: Installing NPM modules using Yarn version 1.3.2

FredyC commented:
Out of curiousity, I tried to build that in separate Netlify project, but the same result sadly. Was hoping it's a problem of cache the Netlify is keeping.

https://app.netlify.com/sites/epic-chandrasekhar-1f8db2/deploys/5cf931f1fd7298202740a260

Wish there was more information, it's seems that some error is being swallowed there.

JustFly1984 commented:
again it seems it installs modules with npm
```6:49:02 PM: Installing NPM modules using NPM version 6.5.0```

FredyC commented:
Uhh, can you show me a link to log that has that? I am seeing Yarn install everywhere.

JustFly1984 commented:
Can you start a new branch from master and commit step by step?

FredyC commented:
Ahh, the `--verbose` of `styleguidist` was the enemy here as Netlify decided to swallow rest of the log with actual errors because of that :) Finally something I can try to fix now :)

```
5:54:28 PM: ✖ ｢atl｣: Checking finished with 2 errors
5:54:28 PM:  FAIL  Failed to compile
5:54:28 PM: ./src/components/addons/MarkerClusterer.tsx
5:54:28 PM: Module not found: Can't resolve '@react-google-maps/marker-clusterer' in '/opt/build/repo/packages/react-google-maps-api/src/components/addons'
5:54:28 PM: build/bundle.71326e60.js from Terser
5:54:28 PM: Unexpected token: operator (!) [build/bundle.71326e60.js:12227,40]
5:54:28 PM: [at-loader] ./src/components/drawing/Marker.tsx:10:27
5:54:28 PM:     TS2307: Cannot find module '@react-google-maps/marker-clusterer'.
5:54:28 PM: [at-loader] ./src/components/addons/MarkerClusterer.tsx:17:8
5:54:28 PM:     TS2307: Cannot find module '@react-google-maps/marker-clusterer'.
```

FredyC commented:
Well, now this is really strange. Your build setup is really using [NPM to install deps]https://app.netlify.com/sites/react-google-maps-api-gatsby-demo/deploys/5cf9395e978a0d0008b5a41d), but my setup [usesYarn](https://app.netlify.com/sites/epic-chandrasekhar-1f8db2/deploys/5cf939ed737aa9c8d8c5d275) over the same sources. 

My build is now passing 😎 It was necessary to build the clusterer first since it's being symlinked there.

My settings are default except I've set these there, so if you can try that as well and rerun the build?

![image](https://user-images.githubusercontent.com/1096340/59048629-88956c80-8886-11e9-81f1-467ac6a0f145.png)

JustFly1984 commented:
I think that environment variables should be setted up in `.netlifyrc`

FredyC commented:
Um, where is the use of that file documented? I know about `netlify.toml`. I will try to use that one instead.

JustFly1984 commented:
7:28:54 PM: Invalid netlify.toml file: Error while decoding file /opt/build/repo/netlify.toml: Near line 3 (last key parsed 'build'): expected a top-level item to end with a newline, comment, or EOF, but got ',' instead
7:28:54 PM: failed during stage 'building site': Error while decoding file /opt/build/repo/netlify.toml: Near line 3 (last key parsed 'build'): expected a top-level item to end with a newline, comment, or EOF, but got ',' instead

FredyC commented:
Well, even with those variables set it's still using NPM. I don't understand why it's behaving like that when with clean project setup it works.

Can you please try to rerun with cache cleanup?

![image](https://user-images.githubusercontent.com/1096340/59050776-79fd8400-888b-11e9-9c17-5969ca132e0f.png)

uriklar commented:
@FredyC @JustFly1984 Maybe this is a good chance to try out:
https://www.netlify.com/products/dev/
?

FredyC commented:
@uriklar Good point, going to try, but I doubt it would help in this case if there is some stale cache on Netlify side.

uriklar commented:
Yeah I had that issue too with netlify 😕 what about linking it to your fork?

FredyC commented:
There is no problem with github linking since it's public repo, but linking to netlify site. I am not on the team so I don't see that site. And as I have shown above, my site is working. Let's not worry about that, it's surely a useful tool, but only with paid plans where you can invite people on the team.

@JustFly1984 Please try that cache clean build, I am fairly certain it will fix it.

JustFly1984 commented:
We got new issue with netlify.toml with gatsby-example deploy
New toml overwrites existing which is used by gatsby-example.
gatsby-example rebuilds at the same time with docs.

FredyC commented:
Don't worry about that for now, I need to see if cleaning cache solves the problem of installation with NPM. I will fix the config later.

JustFly1984 commented:
@FredyC  just remove toml file, I have added env vars to netlify's settings

FredyC commented:
Well, that configuration should be in toml file as well in case it's ever needed to be built on different site (like I needed today). You are stalling for no reason, but I've removed it.

JustFly1984 commented:
It has build successfully with clean cache

FredyC commented:
Uh, where? I see failures... https://app.netlify.com/sites/react-google-maps-api-gatsby-demo/deploys

FredyC commented:
Ah, see it now, I was too fast :) Ok, finally, phew... Let me revert back things I have reverted before and it will be ready for merge.

JustFly1984 commented:
Now I know why styleguidist has failed before without npx!

FredyC commented:
Geez, that's stupid. I've pushed changes and it's installing with NPM again. Can you please try cache reset again?

JustFly1984 commented:
damn, I got it - we are running 2 netlify installs, and they causing racing conditions in checks!

JustFly1984 commented:
Nope, I was wrong

```
8:43:43 PM: Different publish path detected, going to use the one specified in the toml file: 'packages/react-google-maps-api-gatsby-example/public' versus 'packages/react-google-maps-api/styleguide' in the site
```

there is still  netlify.toml file in repo, please remove it completely.

FredyC commented:
Oh, I get it now. You want to keep both netlify installations and each building a different thing? This should be definitely documented somewhere, that's fairly confusing 😅 

Is it a good idea though? It will be causing these race conditions. What about building both as part of one installation and just deploy it under a different path?

FredyC commented:
Ok, I've removed build settings, but kept the env variables just for the sake of other contributors who will want to try the build.

Btw, you can actually remove notifications toward Github from the gatsby site, so it doesn't race condition...

![image](https://user-images.githubusercontent.com/1096340/59055548-5b50ba80-8896-11e9-91df-e0c8a1035d47.png)

FredyC commented:
Wait, you have *Base directory* set for the gatsby example, don't you? That's obvious then because it no longer has `yarn.lock` present there so it uses NPM :D

FredyC commented:
Ok, I have enforced base dir in toml file and now it's finally using Yarn for both! That was really well hidden there.

In the gatsby setup you should be running the `yarn build:gatsby` command now which will build the dependent "api" before it. No need to have `yarn &&` in there, Netlify does the installation by default. Currently, it is failing because of build missing that.

FredyC commented:
That took a bit more stabs, but now both builds are working properly! I've modified scripts temporarily just to show it's both working, but once you modify settings of gatsby site, you can commit root package.json scripts to look like this...

```
    "build": "lerna run build --stream",
    "prebuild:docs": "yarn build:clusterer",
    "build:docs": "cd packages/react-google-maps-api && yarn docs:build && cd ../..",
    "prebuild:gatsby": "yarn build:clusterer",
    "build:gatsby": "cd packages/react-google-maps-api-gatsby-example && yarn build && cd ../..",
    "build:clusterer": "cd packages/react-google-maps-api-marker-clusterer && yarn build && cd ../..",
```

JustFly1984 commented:
I have made next config at netlify for gatsby-example:
<img width="948" alt="Screenshot 2019-06-06 21 41 11" src="https://user-images.githubusercontent.com/2255993/59057860-09af2c80-88a4-11e9-9025-d376dd6829d9.png">

JustFly1984 commented:
and for docs:
<img width="936" alt="Screenshot 2019-06-06 21 44 22" src="https://user-images.githubusercontent.com/2255993/59057994-5135b880-88a4-11e9-916a-351b3458746c.png">

FredyC commented:
Looks correct now, I've pushed changes, should work. Feel free to merge if it does... That was really exhausting, I am done for today 😪
